tradebot/data/cache.py

```
@@ -0,0 +1,332 @@
+
+This module provides efficient local database caching to reduce API calls:
+- DataCache class for storing and retrieving historical market data
+- Cache metadata tracking for management and invalidation
+- Database-backed storage with SQLite for persistence
+- Cache freshness validation based on configurable thresholds
+"""
+
+import pandas as pd
+import logging
+import pickle
+import os
```

Ruff (0.11.9)

`os` imported but unused

Remove unused import: `os`

(F401)